app/src/main/java/com/concordia/mcga/fragments/TransportButtonFragment.java

```diff
@@ -419,4 +458,174 @@ public String getTransportType() {
+            sqlTableColumnIndex += 1;
+        else
+            sqlTableColumnIndex -= 1;
+
+        return sqlTableColumnIndex;
+    }
+
+    /**
+     * Returns the appropriate column index for the shuttle,
+     * based on currentDay and the current Campus.
+     *
+     * @param currentDay
+     * @return
+     */
+    private int getShuttleColumnIndex(int currentDay) {
+        int sqlTableColumnIndex = 0;
+
+        if (currentCampus == Campus.LOY)
+            sqlTableColumnIndex += 2; //LOYtoSGW are columns 2,3
+
+        if (currentDay > 4)
+            sqlTableColumnIndex += 1; //Friday, Weekends are columns 1,3
+
+        return sqlTableColumnIndex;
+    }
+
+    /**
+     * Shuttle service is only available from Monday to Friday
+     */
+    private boolean isShuttleAvailable(int day) {
+        return (day > 0 && day < 6);
+    }
+
+    /**
+     * This is a parsing method to extract time in minutes from a String.
+     *
+     * @param hourminute "HH:mm"
+     * @return minutes
+     */
+    private int getMinutesFromTimeString(String hourminute) {
+        String[] tokens = hourminute.split(":");
+        if (tokens[0].contains("-")) {
+            return -1;
+        } else {
+            return Integer.valueOf(tokens[0]) * 60 + Integer.valueOf(tokens[1]);
+        }
+    }
+
+    /**
+     * Calculates the time in minutes until the next Concordia Shuttle
+     * departure. The method iterates over the list and tries to find a
+     * match for the next available departure.
+     *
+     * @param shuttleSchedule
+     * @param columnIndex
+     * @param currentTimeInMinutes
+     * @return returns the time until the next departure, or -1 if reached end of list
+     */
+    private int calculateTimeToNextShuttle(String[][] shuttleSchedule, int columnIndex, int currentTimeInMinutes) {
+        //Scans the list from the beginning, for current day
+        for (int rowIndex = 0; rowIndex < shuttleSchedule.length; rowIndex++) {
+            int scheduleSlotInMinutes = getMinutesFromTimeString(shuttleSchedule[rowIndex][columnIndex]);
+            //There are "-" in the schedule for empty slots
+            if (scheduleSlotInMinutes == -1) {
+                continue;
+            }
+
+            int timeDifference = scheduleSlotInMinutes - currentTimeInMinutes;
+            if (timeDifference > 0) {
+                return timeDifference;
+            }
+        }
+        //Current time is after last shuttle for day has departed
+        return -1;
+    }
+
+    /**
+     * Calculates the number of whole days until the
+     * next day that has Concordia Shuttle Service.
+     *
+     * @param currentDay
+     * @return
+     */
+    private int calculateNumberOfWholeDaysToNextShuttle(int currentDay) {
+
+        //This should never occur.
+        if (currentDay < 0 || currentDay > 7)
+            return -1;
+
+        //Friday, have to wait for Saturday, Sunday
+        if (currentDay == 5)
```

Comment:
You might want to have some sort of enum with these days associated these numbers.